lib/bannerFactory.js

```diff
@@ -33,21 +35,83 @@ class BannerFactory extends BaseFactory {
      * @param {String}  config.createdBy           The username of the associated user
      * @param {Boolean} [config.isActive=false]    Whether the banner is active
      * @param {String}  [config.type='info']       Type of banner (info|warn|etc)
+     * @param {String}  [config.scope='GLOBAL']    Scope of the banner (GLOBAL|PIPELINE|BUILD)
      * @memberof BannerFactory
      */
-    create(config) {
+    async create(config) {
         if (!config.type) {
             config.type = 'info';
         }
         if (!config.isActive) {
             config.isActive = false;
         }
+        if (!config.scope) {
+            config.scope = 'GLOBAL';
+            config.scopeId = null;
+        }
+        if ((config.scope === 'PIPELINE' || config.scope === 'BUILD') && !config.scopeId) {
+            throw new Error(`scopeId is required when scope is ${config.scope}`);
+        }
+        if (config.scope === 'PIPELINE') {
+            const pipeline = await PipelineFactory.getInstance().get(config.scopeId);
+
+            if (!pipeline) {
+                throw new Error(`Pipeline ${config.scopeId} does not exist`);
+            }
+        }
+        if (config.scope === 'BUILD') {
+            const build = await BuildFactory.getInstance().get(config.scopeId);
+
+            if (!build) {
+                throw new Error(`Build ${config.scopeId} does not exist`);
+            }
+        }
```

Comment:
Can we refactor the duplicate code block into a reusable function to validate the `scopeId` by choosing the appropriate factory based on the `scope`